File: java/parsers/src/test/java/jp/co/teruuu/parser/json/JObjectParserTest.java
package jp.co.teruuu.parser.json;

import jp.co.teruuu.parser.common.ParseResult;
import org.junit.Test;

public class JObjectParserTest {

  @Test
  public void test() {
    JParser parser = JParser.object();
    ParseResult result1 = parser.parse("{}");
    assert(result1 instanceof ParseResult.Success);

    ParseResult result2 = parser.parse("{\"a\": 1}");
    assert(result2 instanceof ParseResult.Success);

    ParseResult result3 = parser.parse("{\"a\": 1, \"c\": []}");
    assert(result3 instanceof ParseResult.Success);

    ParseResult result4 = parser.parse("{\"a\": 1, \"c\": [{\"d\": 345}]}");
    assert(result4 instanceof ParseResult.Success);
  }
}

File: scala/parsers/src/test/scala/jp/co/teruuu/parser/json/JObjectParserTest.scala
package jp.co.teruuu.parser.json

import jp.co.teruuu.parser.common.ParseOk
import jp.co.teruuu.parser.json.`type`._
import org.scalatest.FlatSpec

class JObjectParserTest extends FlatSpec{

  "parse" should "successt" in {
    val parser = JObjectParser

    val result1 = parser.parse(" { } ")
    assert(result1.isInstanceOf[ParseOk[JObject]])
    assert(result1.asInstanceOf[ParseOk[JObject]].value.equals(JObject(Map.empty[JString, JValue])))

    val result2 = parser.parse("{\"a\": 1, \"c\": [{\"d\": 345}]}")
    assert(result2.isInstanceOf[ParseOk[JObject]])
    assert(result2.asInstanceOf[ParseOk[JObject]].value.equals(JObject(
      Map(JString("a") -> JNumber(1, 0, 0), JString("c") -> JArray(List(JObject(Map(JString("d") -> JNumber(345, 0, 0))))))
    )))

  }

}
